tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/paging/PageFilterTest.java

```diff
@@ -0,0 +1,163 @@
+      ActiveMQServer server = createServer(true, isNetty());
+
+      AddressSettings defaultSetting = new AddressSettings().setPageSizeBytes(10 * 1024).setMaxSizeMessages(10);
+
+      server.getAddressSettingsRepository().addMatch("#", defaultSetting);
+
+      return server;
+   }
+
+   @Test
+   public void testPageFilter() throws Exception {
+
+      ConnectionFactory connectionFactory = CFUtil.createConnectionFactory("CORE", "tcp://localhost:61616");
+
+      int numberOfMessages = 3 * 24;
+      String filter = "partial=true";
+      String negfilter = "partial=false";
+
+      server.addAddressInfo(new AddressInfo(getName()).addRoutingType(RoutingType.MULTICAST));
+
+
+
+      try (Connection connection = connectionFactory.createConnection();
+           Connection otherConnection = connectionFactory.createConnection()) {
+         Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+         Topic topic = session.createTopic(getName());
+         session.createSharedDurableConsumer(topic, "wholeThing", negfilter);
+         MessageConsumer partialThing = session.createSharedDurableConsumer(topic, "partialThing", filter);
+
+         final String oneThousandCharacters;
+         {
+            StringBuilder builder = new StringBuilder(1000);
+            for (int i = 0; i < 1000; i++) {
+               builder.append('a');
+            }
+            oneThousandCharacters = builder.toString();
+         }
+
+         MessageProducer producer = session.createProducer(topic);
+
+
+         Session txsession = otherConnection.createSession(true, Session.SESSION_TRANSACTED);
+         MessageProducer txProducer = txsession.createProducer(topic);
+         for (int i = 0; i < 100; i++) {
+            Message message = session.createTextMessage(oneThousandCharacters + i);
+            message.setIntProperty("i", i);
+            message.setBooleanProperty("partial", i % 2 == 0);
+            producer.send(message);
+            if (i % 4 == 0) {
+               Message txMessage = session.createTextMessage("ignored");
+               txMessage.setBooleanProperty("partial", false);
+               txMessage.setBooleanProperty("ignored", true);
+               txProducer.send(txMessage);
+            }
+         }
+
+         connection.start();
+
+         int receivedI = 0;
+         do {
+            Message message = partialThing.receive(1000);
+            Assert.assertNotNull(message);
+            receivedI = message.getIntProperty("i");
+         }
+         while (receivedI < 10);
+
+         server.stop();
+      }
+
+      server.start();
+
+      try (Connection connection = connectionFactory.createConnection()) {
+         Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+         Topic topic = session.createTopic(getName());
+         connection.start();
+         MessageConsumer consumerWhole = session.createSharedDurableConsumer(topic, "partialThing", filter);
```

Comment:
consumerWhole using the "partialThing" sub seems confusing (as opposed to e.g the "wholeThing" sub that was created but not used...and which itself doesnt get everything anyway). Perhaps some clearer naming?